### ISSUE TITLE
Recursive upload directory causes infinite loop

### ISSUE THREAD

**Issue by danielbardsley** [gilmarsquinelato/serverless-spa#1]
If the source directory contains sub directories then the plugin will get stuck in an infinite loop.

```return stats.isDirectory() ? this._uploadDirectory() : this._uploadFile(path);```

The line shown above will repeatedly call _uploadDirectory() with no parameters and get stuck in an loop. This can be rectified by changing the code in the _uploadDirectory method to be as follows:

```
_uploadDirectory(recursivePath) {
    let directoryPath = this.distFolder;
    if (JSON.stringify(recursivePath) !== "{}") {
        directoryPath = recursivePath
    }
    let readDirectory = _.partial(fs.readdir, directoryPath);

    async.waterfall([readDirectory, (files) => {
      files = _.map(files, (file) => path.join(directoryPath, file));

      async.each(files, (path) => {
        fs.stat(path, (err, stats) => {
          return stats.isDirectory() ? this._uploadDirectory(path) : this._uploadFile(path);
        });
      });
    }]);
  }
```
I am happy to create a pull request to fix this if desired.

Thanks.

**Comment by gilmarsquinelato** [comment]
Hi @danielbardsley thank you for the issue, and pull requests are always welcome =).

**Comment by gilmarsquinelato** [comment]
Closing issue due to inactivity.

**Comment by ivands** [comment]
This is broken again.
Seems like the PR didn't get merged.

**Comment by danielbardsley** [comment]
I've added a pull request to resolve the issue: https://github.com/gilmarsquinelato/serverless-spa/pull/5

**Comment by ivands** [comment]
Thanks @danielbardsley!
Lets hope it will be added soon

**Comment by gilmarsquinelato** [comment]
Hello guys!
Sorry for the delay, and thanks for the PR.
I've accepted and published the v1.2.3.

**Comment by ivands** [comment]
Thanks and great work btw @gilmarsquinelato
